### PR TITLE
Pay metadata considerations for `DelegateVersion 1.1`

### PR DESCRIPTION
**Pull request by johnnyd-eth**
## What does this PR do and why?

No UI changes. Another step for https://github.com/jbx-protocol/juice-interface/issues/2794

## Screenshots or screen recordings

_If applicable, provide screenshots or screen recordings to demonstrate the changes._

## Acceptance checklist

- [ ] I have evaluated the [Approval Guidelines](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#approval-guidelines) for this PR.
- [ ] I have tested this PR in [all supported browsers](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#supported-browsers).
- [ ] I have tested this PR in dark mode and light mode (if applicable).

### CODE REVIEW

File: src/components/v2v3/V2V3Project/V2V3PayButton/V2V3ConfirmPayModal/V2V3ConfirmPayModal.tsx [hunk 1]
```diff
@@ -135,10 +140,18 @@ export function V2V3ConfirmPayModal({
     } = form.getFieldsValue()
 
     const txBeneficiary = beneficiary ?? userAddress
-    const delegateMetadata = encodeJB721DelegatePayMetadata({
-      ...(payProjectForm?.payMetadata as JB721DelegatePayMetadata),
-      ...payMetadataOverrides(projectId),
-    })
+
+    const delegateMetadata =
+      nftContractVersion === '1' // old delegate v1
```
Comment (tomquirk) [inline src/components/v2v3/V2V3Project/V2V3PayButton/V2V3ConfirmPayModal/V2V3ConfirmPayModal.tsx:145]: **suggestion** pretty sure there is a constant for `'1'` right?

File: src/components/v2v3/V2V3Project/V2V3PayButton/V2V3ConfirmPayModal/V2V3ConfirmPayModal.tsx [hunk 2]
```diff
@@ -135,10 +140,18 @@ export function V2V3ConfirmPayModal({
...
+        ? encodeJB721DelegateV1PayMetadata({
+            ...(payProjectForm?.payMetadata as JB721DelegateV1PayMetadata),
+            ...payMetadataOverrides(projectId),
+          })
+        : // new delegate v1.1
+          encodeJB721DelegateV1_1PayMetadata({
```
Comment (tomquirk) [inline src/components/v2v3/V2V3Project/V2V3PayButton/V2V3ConfirmPayModal/V2V3ConfirmPayModal.tsx:151]: **issue** i think we should specifically only do this if      `nftContractVersion === '1.1'`. Might save our ass in the future.

**Review by tomquirk — APPROVED** [review]
@johnnyd-eth pre-approving. pls see comments, and also would be good to allow underscores in typenames in tslint (or whatever) if possible, to avoid `V11`.

File: src/components/v2v3/V2V3Project/V2V3PayButton/V2V3ConfirmPayModal/V2V3ConfirmPayModal.tsx (hunk 2 above)
Comment (johnnyd-eth) [reply src/components/v2v3/V2V3Project/V2V3PayButton/V2V3ConfirmPayModal/V2V3ConfirmPayModal.tsx:151]: Good call, done

File: src/components/v2v3/V2V3Project/V2V3PayButton/V2V3ConfirmPayModal/V2V3ConfirmPayModal.tsx (hunk 1 above)
Comment (johnnyd-eth) [reply src/components/v2v3/V2V3Project/V2V3PayButton/V2V3ConfirmPayModal/V2V3ConfirmPayModal.tsx:145]: Yep, there is. Thanks

**Comment by johnnyd-eth** [comment]
> @johnnyd-eth pre-approving. pls see comments, and also would be good to allow underscores in typenames in tslint (or whatever) if possible, to avoid `V11`.

Best I could do was allow only upper-case with underscores as well. These are the only rules available: https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/naming-convention.md#format-options

**Pull request merged**